eZ/Publish/Core/Repository/BookmarkService.php

```diff
@@ -105,9 +105,17 @@ public function loadBookmarks(int $offset = 0, int $limit = 25): BookmarkList
         if ($list->totalCount > 0) {
             $bookmarks = $this->bookmarkHandler->loadUserBookmarks($currentUserId, $offset, $limit);
 
+            // https://jira.ez.no/browse/EZP-30120 
+            // The bookmark page should not crash if the user has content in his bookmark that he is not allowed to read. 
             $list->items = array_map(function (Bookmark $bookmark) {
                 return $this->repository->getLocationService()->loadLocation($bookmark->locationId);
```

Comment:
can / should change this to call `loadLocationList()`, it will filter out items you don't have access to (NotFound and Unauthorized), and do the lookup in one go instead of several in loop

aka, first iterate or array_map to just get location ids, then call locaLocationList and set directly on `->items`

Reply:
The question of what to do with inaccessible elements. 
Do we hide them silently. In this case, they should not be counted. (There's a small bookmark counter at the top of the list.)

Or you can make them appear in the list. 
In this case two solutions. Either the name of the object is displayed. Either we don't post it. 
In any case, the user must be allowed to delete these bookmarks.

Reply:
That's a good point, and the answer to that might lay in revising the whole design of loadBookmarks(). So maybe more for @adamwojs for instance to give input on.

Reply:
> Do we hide them silently. In this case, they should not be counted. (There's a small bookmark counter at the top of the list.)

Exactly, we need take permissions into account in the `totalCount` value as well, otherwise the pagination in `ezplatform-admin-ui` will be broken. 

First solution: use `SearchService::findLocation` to load bookmarks with new dedicated `IsBookmarked` criterion.  

Second solution: introduce `Placeholder Object` which will be returned in case of insufficient permissions. I mean something like this:

```php
/**
 * @copyright Copyright (C) eZ Systems AS. All rights reserved.
 * @license For full copyright and license information view LICENSE file distributed with this source code.
 */
declare(strict_types=1);

namespace eZ\Publish\API\Repository\Values\Content;

/**
 * This class represents a location which is unreachable for current user for some reason: usually because of insufficient permissions.
 */
abstract class UnreachableLocation extends Location 
{
     // ...
}
```

This requires adjusting other PAPI services and UI to work with it (what should happen when `UnreachableLocatio`n will be passed as parameter of `\eZ\Publish\Core\Repository\LocationService::getLocationChildCount` ?).

Actually this is quite common problem named "Filtering list of objects by permissions to related content/location" :wink: .

Reply:
I would maybe go for 1, but then aim to as soon  as possible get buy in from PM to prioritise LocationService and ContentService->filter() _(ref closed PR from Petar, also ref PM's product backlog story about being able to query persistence)_.